### ISSUE TITLE
Overrides in Sync Patterns: expand block support

### ISSUE THREAD

**Issue by liviopv** [WordPress/gutenberg#59607]
## What problem does this address?
Users should be able to overrides more blocks that are can hold actual content, not just Paragraph/Heading/Image/Button

## What is your proposed solution?
Some ideas:

- List Block > List Items
- Quote Block: citation
- Gallery > Image
- Verse
- Preformatted
- File
- Video
- Embed (and variations)
- Nested blocks inside: Group, Cover, Column, Media & Text

**Comment by gziolo** [comment]
@SantosGuillamot, that’s also related to Block Bindings.

**Comment by SantosGuillamot** [comment]
The idea is to add support for more blocks progressively to ensure everything works as expected. It's great to know which ones feel more important to have that in mind.

**Comment by fabiankaegy** [comment]
Should we also add "custom blocks" to the list here? As a developer I would love to be able to expose the block connections / pattern overrides feature in custom blocks I need to build.

**Comment by SantosGuillamot** [comment]
Yes, custom blocks are also part of the roadmap, although it will probably take more time. They depend on the HTML API supporting any HTML tag and they will need a proper opt-in mechanism. I've added it to the [block bindings API epic](https://github.com/WordPress/gutenberg/issues/53300).

**Comment by gziolo** [comment]
From the list above the Verse and Preformatted blocks should be close in implementation to the Paragraph and Heading block so if someone would like to give it a try, @SantosGuillamot and I can help with reviews.

**Comment by dannyreaktiv** [comment]
The Group block would be a great way to provide an open area for inner-blocks.

**Comment by annezazu** [comment]
>The Group block would be a great way to provide an open area for inner-blocks.

@dannyreaktiv  you'll dig this issue on that exact topic: https://github.com/WordPress/gutenberg/issues/61225

**Comment by dannyreaktiv** [comment]
> > The Group block would be a great way to provide an open area for inner-blocks.
> 
> @dannyreaktiv you'll dig this issue on that exact topic: #61225

Wow. Thank you!

**Comment by asafm7** [comment]
I believe the Block Binding support for the Embed block can be useful.

Is this the best place to suggest it, or is there a dedicated thread?

**Comment by SantosGuillamot** [comment]
I believe we can use this same conversation to gather feedback on the most important blocks 🙂

**Comment by bacoords** [comment]
Overrides for the Cover Block Image: https://github.com/WordPress/gutenberg/issues/63763

**Comment by jhmonroe** [comment]
Because of this limitation and the way the Details block is built, there's also no way for someone to use this feature in a reusable block.

A major use case is the attached image where I'm building a page of dozens of speaker bios on a page that are collapsed by default so the page isn't overwhelming.

Aside: Maybe this ship has sailed, but if the <details> block was treated more like a Group block wouldn't this enable someone to edit the first text as a heading or paragraph instead? (and thus the "group" could be the reusable block and all the content would be editable separately. I see now it currently outputs that first bit of text as <summary> although it styles it the same as normal paragraph text.

Attached screenshot is from v6.6.1

On Pattern Editor:
![Screen Shot 2024-07-25 at 11 07 54 PM](https://github.com/user-attachments/assets/8fcd3183-6bc1-4d8a-a968-a3d7b1fd7aa2)

On Page:
![Screen Shot 2024-07-25 at 11 16 23 PM](https://github.com/user-attachments/assets/6f08df85-2959-4545-9ad2-65da13ab4b10)

**Comment by gziolo** [comment]
@fabiankaegy created a related issue explaining how we could open support for custom blocks that are rendered dynamically with `render_callback` on the server:
- https://github.com/WordPress/gutenberg/issues/64870

More ideas to core blocks that act like containers:
- https://github.com/WordPress/gutenberg/issues/61225
- https://github.com/WordPress/gutenberg/issues/59606
